run_foundry.py

```diff
@@ -0,0 +1,478 @@
+                definition_dir=krun.definition_dir,
+                output=KRunOutput.JSON,
+                parser='cat',
+                term=True,
+                check=False,
+                pipe_stderr=True,
+            )
+
+
+def parse_proc_res(proc_res: CompletedProcess) -> KInner:
+    return kast_term(json.loads(proc_res.stdout), KInner)
+
+
+def run_config_and_check_empty(krun: KRun, conf: KInner) -> tuple[KInner, KInner, dict[str, KInner]]:
+    final_conf = parse_proc_res(run_config(krun, conf))
+    sym_conf, subst = split_config_from(final_conf)
+    k_cell = subst['K_CELL']
+    if not isinstance(k_cell, KSequence) or k_cell.arity != 0:
+        print(krun.pretty_print(subst['VMOUTPUT_CELL']), file=sys.stderr)
+        print(krun.pretty_print(subst['K_CELL']), file=sys.stderr)
+        raise ValueError(f'k cell not empty:\n { krun.pretty_print(final_conf) }')
+
+    return final_conf, sym_conf, subst
+
+
+def run_test(krun: KRun, sym_conf, init_subst, endpoint, args):
+    step = {
+        'tx': {
+            'from': 'address:k',
+            'to': 'sc:k-test',
+            'function': endpoint,
+            'value': '0',
+            'arguments': args,
+            'gasLimit': '5,000,000,000',
+            'gasPrice': '0',
+        },
+        'expect': {'status': '0'},
+    }
+    tx_steps = KSequence([KApply('setExitCode', [KInt(1)])] + get_steps_sc_call(step) + [KApply('setExitCode', [KInt(0)])])
+    
+    subst = init_subst.copy()
+    subst['K_CELL'] = tx_steps
+    conf_with_steps = Subst(subst)(sym_conf)
+    
+    proc_res = run_config(krun, conf_with_steps)
+    if proc_res.returncode:
+        raise RuntimeError(f'Run failed: {args}')
+    
+# Test metadata
+TEST_PREFIX = 'test_'
+
+
+def get_test_endpoints(test_dir: str) -> Mapping[str, tuple[str, ...]]:
+    abi_path = glob.glob(test_dir + '/output/*.abi.json')
+    # TODO this loads the first wasm file in the directory. what if there are multiple wasm files?
+    if abi_path:
+        abi_path = abi_path[0]
+    else:
+        raise ValueError(f'ABI file not found: {test_dir}/output/?.abi.json')
+
+    with open(abi_path, 'r') as f:
+        abi_json = json.load(f)
+
+    endpoints = {}
+    for endpoint in abi_json['endpoints']:
+        name = endpoint['name']
+        if not name.startswith(TEST_PREFIX):
+            continue
+
+        inputs = tuple(i['type'] for i in endpoint['inputs'])
+
+        endpoints[name] = inputs
+
+    return endpoints
+
+
+# Hypothesis strategies
+
+
+def type_to_strategy(typ: str):
+    if typ == 'BigUint':
+        return integers(min_value=0).map(str)
+    if typ == 'u32':
+        return integers(min_value=0, max_value=4294967295).map(str)
+    raise TypeError(f'Cannot create random {typ}')
+
+
+def arg_types_to_strategy(types):
+    strs = (type_to_strategy(t) for t in types)
+    return tuples(*strs)
+
+
+def test_with_hypothesis(krun, sym_conf, init_subst, endpoint, arg_types):
```

Comment:
Given that these arguments do not have types - newer pyk-based projects check for a lot of things, including that all variables have types (implicit or explicit), and that types match / are compatible. Maybe it would be worth adding something like that here, too (perhaps in a separate PR).

See, e.g., this:
https://github.com/runtimeverification/pyk/blob/master/Makefile#L67-L92

Reply:
Added type annotations in this PR, will create a poetry package using the cookiecutter template in the next PR.